.github/workflows/bundle-size.yml

```diff
@@ -0,0 +1,170 @@
+
+      - name: Measure the baseline size of the Vite project
+        if: steps.restore-cache-baseline.outputs.cache-hit != 'true'
+        run: |
+          cd bundle-size
+          npm run build
+          mv output.txt output-baseline.txt
+          cat output-baseline.txt
+
+      - name: Cache the baseline size
+        uses: actions/cache/save@v3
+        if: steps.restore-cache-baseline.outputs.cache-hit != 'true'
+        with:
+          path: bundle-size/output-baseline.txt
+          key: ${{ steps.restore-cache-baseline.outputs.cache-primary-key }}
+
+      # ------- Base branch size measurement ------------------------------
+
+      - name: Checkout the base branch
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.base.sha }}
+          path: 'components'
+
+      - name: Restore the cache for the base branch size measurement
+        uses: actions/cache/restore@v3
+        id: restore-cache-basebranch
+        with:
+          path: bundle-size/output-basebranch.txt
+          key: bundlesize-${{ hashFiles('bundle-size/build.js','bundle-size/index.html','bundle-size/main-with-cloudscape.jsx','bundle-size/package-lock.json','bundle-size/package.json') }}-${{ github.event.pull_request.base.sha }}
+
+      - name: Measure the bundle size of the base branch
+        if: steps.restore-cache-basebranch.outputs.cache-hit != 'true'
+        run: |
+          cd components
+          npm i --force
+          npm run quick-build
+
+          cd ../bundle-size
+          rm -rf node_modules/@cloudscape-design/components
+          cp -r $GITHUB_WORKSPACE/components/lib/components node_modules/@cloudscape-design/components
+          cp main-with-cloudscape.jsx main.jsx
+          npm run build
+
+          mv output.txt output-basebranch.txt
+          cat output-basebranch.txt
+
+      - name: Cache the base branch size
+        uses: actions/cache/save@v3
+        if: steps.restore-cache-basebranch.outputs.cache-hit != 'true'
+        with:
+          path: bundle-size/output-basebranch.txt
+          key: ${{ steps.restore-cache-basebranch.outputs.cache-primary-key }}
+
+      # ------- Pull request size measurement -----------------------------
+
+      - name: Checkout the PR branch
+        uses: actions/checkout@v3
+        with:
+          path: 'components'
+
+      - name: Measure the bundle size of this PR
+        run: |
+          cd components
+          npm i --force
+          npm run quick-build
+
+          cd ../bundle-size
+          rm -rf node_modules/@cloudscape-design/components
+          cp -r $GITHUB_WORKSPACE/components/lib/components node_modules/@cloudscape-design/components
+          cp main-with-cloudscape.jsx main.jsx
+          npm run build
+
+          mv output.txt output-pr.txt
+          cat output-pr.txt
+
+      - name: Update the commit status with calculated results
+        uses: actions/github-script@v6
+        with:
+          script: |
```

Comment:
That much of JS inline looks hard to read and maintain. Also there is a risk of injections (see our past issues on https://github.com/cloudscape-design/.github/pull/29).

I am on the fence about "Create a pending status on the commit" steps (short and might be okay). But the long script definitely deserves a separate file